lib/registry/src/config.rs

```diff
@@ -0,0 +1,329 @@
+                        current: format_graphql(&s.url),
+                        tokens: map,
+                    })
+                }
+            }
+            Registries::Multi(m) => {
+                m.tokens.insert(format_graphql(registry), token.to_string());
+                if update_current_registry == UpdateRegistry::Update {
+                    m.current = format_graphql(registry);
+                }
+                Registries::Multi(m.clone())
+            }
+        };
+        *self = new_map;
+    }
+}
+
+impl PartialWapmConfig {
+    /// Save the config to a file
+    #[cfg(not(feature = "integration_tests"))]
+    pub fn save(&self) -> anyhow::Result<()> {
+        use std::{fs::File, io::Write};
+        let path = Self::get_file_location().map_err(|e| anyhow::anyhow!("{e}"))?;
+        let config_serialized = toml::to_string(&self)?;
+        let mut file = File::create(path)?;
+        file.write_all(config_serialized.as_bytes())?;
+        Ok(())
+    }
+
+    /// A mocked version of the standard function for integration tests
+    #[cfg(feature = "integration_tests")]
+    pub fn save(&self) -> anyhow::Result<()> {
+        let config_serialized = toml::to_string(&self)?;
+        crate::integration_tests::data::RAW_CONFIG_DATA.with(|rcd| {
+            *rcd.borrow_mut() = Some(config_serialized);
+        });
+
+        Ok(())
+    }
+
+    pub fn from_file() -> Result<Self, String> {
+        let path = Self::get_file_location()?;
+
+        match std::fs::read_to_string(&path) {
+            Ok(config_toml) => {
+                toml::from_str(&config_toml).map_err(|e| format!("could not parse {path:?}: {e}"))
+            }
+            Err(_e) => Ok(Self::default()),
+        }
+    }
+
+    pub fn get_current_dir() -> std::io::Result<PathBuf> {
+        #[cfg(target_os = "wasi")]
+        if let Some(pwd) = std::env::var("PWD").ok() {
+            return Ok(PathBuf::from(pwd));
+        }
+        std::env::current_dir()
+    }
+
+    pub fn get_folder() -> Result<PathBuf, String> {
+        Ok(
+            if let Some(folder_str) = env::var("WASMER_DIR").ok().filter(|s| !s.is_empty()) {
+                let folder = PathBuf::from(folder_str);
+                std::fs::create_dir_all(folder.clone())
+                    .map_err(|e| format!("cannot create config directory: {e}"))?;
+                folder
+            } else {
+                #[allow(unused_variables)]
+                let default_dir = Self::get_current_dir()
+                    .ok()
+                    .unwrap_or_else(|| PathBuf::from("/".to_string()));
+                #[cfg(feature = "dirs")]
+                let home_dir =
+                    dirs::home_dir().ok_or(GlobalConfigError::CannotFindHomeDirectory)?;
+                #[cfg(not(feature = "dirs"))]
+                let home_dir = std::env::var("HOME")
+                    .ok()
+                    .unwrap_or_else(|| default_dir.to_string_lossy().to_string());
```

Comment:
Is it worth just using the `dirs` crate unconditionally and dropping the feature flag?

Reply:
okay